File: 1620-check-if-array-pairs-are-divisible-by-k/1620-check-if-array-pairs-are-divisible-by-k.c
bool canArrange(int* arr, int arrSize, int k) {
    int rem_cnt[k];
    for (int i = 0; i < k; i++) rem_cnt[i] = 0;
    for (int i = 0; i < arrSize; i++) rem_cnt[(arr[i] % k + k) % k]++;
    int rem;
    for (int i = 0; i < arrSize; i++) {
        rem = (i % k + k) % k;

        if (!rem) {
            if (rem_cnt[rem] & 1) return false; 
        }
        else if (rem_cnt[rem] != rem_cnt[k - rem]) return false;
    }
    return true;
}

File: 1620-check-if-array-pairs-are-divisible-by-k/1620-check-if-array-pairs-are-divisible-by-k.kt
class Solution {
    fun canArrange(arr: IntArray, k: Int): Boolean {
        val rem_cnt: IntArray = IntArray(k) { 0 };
        arr.forEach { ele -> rem_cnt[(ele % k + k) % k]++ };
        var rem: Int;
        for (ele in arr) {
            rem = (ele % k + k) % k;
            if (rem == 0) {
                if (rem_cnt[rem] % 2 == 1) return false;
            } else if (rem_cnt[rem] != rem_cnt[k - rem]) return false;
        }
        return true;
    }
}